app/bundles/LeadBundle/Controller/ListController.php

```diff
@@ -827,6 +827,133 @@ public function viewAction($objectId)
         ]);
     }
 
+    /**
+     * Get the permission base from the model.
+     *
+     * @return string
+     */
+    protected function getPermissionBase()
+    {
+        return $this->getModel('lead.list')->getPermissionBase();
+    }
+
+    /**
+     * @return ListModel
+     */
+    protected function getListModel()
```

Comment:
Thanks a lot for this PR @bkomel! Overall the code looks good, but it could benefit from some explicit return types here and there. E.g. this one would become `protected function getListModel(): ListModel`, you can then remove the `@return ListModel` above. We're trying to have as much code as possible in that style to prevent type-related bugs. 

Let me know if you need help with that, happy to help!

Reply:
@bkomel will you be able to look at these suggestions?